### PR TITLE
Mongoose

### PR DESCRIPTION
**Pull request by HazemKhaled**
## :memo: Description

1. Adding Mongoose and remove Mongo adapter
2. Add example Model

### :gem: Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js
// services/products.service.ts
import { ProductDocument, ProductModel } from '../utilities/models';
...
mixins: [new DbService('Products').start<ProductDocument>(ProductModel)]
...
```
```js
// utilities/models.product.ts
import mongoose, { Schema, Document } from 'mongoose';

import { Product } from '../types';

const ProductSchema = new Schema<Product>({
  name: { type: String, required: true, unique: true },
  category: { type: String },
  price: { type: Number, required: true },
});

export interface ProductDocument extends Document<Product> {}

export const ProductModel = mongoose.model<ProductDocument>(
  'Product',
  ProductSchema
);
``` 

## :vertical_traffic_light: How Has This Been Tested?

- [ ] `NODE_ENV=test npm run dev` on node `v14.15.3`

## :checkered_flag: Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] **I have added tests that prove my fix is effective or that my feature works**
- [ ] **New and existing unit tests pass locally with my changes**
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: services/products.service.ts [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-   * Actions
-   */
   actions: {
     create: {
       rest: 'POST /',
       auth: [],
       cache: {},
-      async handler(ctx): Promise<Product> {
+      async handler(ctx: Context<Product>) {
         return this.adapter
```
Comment (dhavalsoni9989) [inline services/products.service.ts:23]: In mp we have used ctx.call for insert the data 
For Eg. ctx
          .call<Coupon, Coupon>(
            'coupons.create',
            ctx.params
          )
          .then(res => {
            return this.normalizeId(res);
          })

File: services/products.service.ts [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
 import DbService from '../utilities/mixins/mongo.mixin';
 import { MpError } from '../utilities/adapters';
 import { Product } from '../utilities/types';
+import { ProductDocument, ProductModel } from '../utilities/models';
 
 const ProductsService: ServiceSchema = {
   name: 'products',
-  mixins: [DbService('Products'), ProductsValidation, ProductsOpenapi],
+  mixins: [
+    new DbService('Products').start<ProductDocument>(ProductModel),
```
Comment (neer14) [inline services/products.service.ts:12]: why start here?

Comment (HazemKhaled) [reply services/products.service.ts:12]: check the new mixins

File: services/products.service.ts (hunk 1 above)
Comment (HazemKhaled) [reply services/products.service.ts:23]: it's just example, let me make it more clear

**Pull request merged**